### PR TITLE
Spanish language update

### PR DESCRIPTION
**Pull request by Underforest**
Huge improvements in spanish localisation, plus added missing strings and fixed outdated ones. Updated with Feb 11, 2018 English locale as base.

This should be now ready to go, but any suggestion or fix is welcome~

### CODE REVIEW

File: resources/lang/es/beatmaps.php [hunk 1]
```diff
@@ -83,13 +89,44 @@
...
+            'mapper_notes' => 'Notas',
+            'mine' => 'Mi autoría',
             'pending' => 'Pendiente',
             'praises' => 'Elogios',
             'resolved' => 'Resuelto',
+            'total' => 'Todo',
+        ],
+
+        'status-messages' => [
+            'approved' => '¡Este beatmap fue aprobado en :date!',
```
Comment (ColdVolcano) [inline resources/lang/es/beatmaps.php]: fue aprobado el :date

File: resources/lang/es/beatmaps.php [hunk 2]
```diff
@@ -83,13 +89,44 @@
...
+            'graveyard' => 'Este mapa no ha sido actualizado desde el :date y pudo haber sido abandonado por el creador...',
+            'loved' => '¡Este beatmap fue marcado como amado el :date!',
+            'ranked' => '¡Este beatmap ha sido rankeado el :date!',
```
Comment (ColdVolcano) [inline resources/lang/es/beatmaps.php]: fue rankeado el :date

File: resources/lang/es/beatmaps.php [hunk 3]
```diff
@@ -83,13 +89,44 @@
...
         ],
+
+    ],
+
+   'hype' => [
+        'button' => '¡Hypear este beatmap!',
+        'button_done' => '¡Hypeado!',
+        'confirm' => '¿Estás seguro? Esto utilizará uno de tus :n hypes restantes y no podrás deshacerlo.',
+        'explanation' => '¡Hypea este beatmap para hacerlo más visible para la nominación y el ranking!',
+        'explanation_guest' => 'Inicia sesión y hypea este beatmap para hacerlo más visible para la nominación y el ranking!',
```
Comment (ColdVolcano) [inline resources/lang/es/beatmaps.php]: ¡ 🤔

File: resources/lang/es/beatmaps.php [hunk 4]
```diff
@@ -100,18 +137,22 @@
         'incorrect_state' => 'Error al realizar esa acción, intenta recargando la página.',
         'nominate' => 'Nominar',
         'nominate_confirm' => '¿Nominar este beatmap?',
+        'nominated_by' => 'nominado por :users',
         'qualified' => 'Se estima que será rankeado :date, si no se encuentra ningún problema.',
-        'qualified_soon' => 'Se estima que será rankeado pronto, si no se encuentra ningún problema.',
+        'qualified_soon' => 'Se estima que será rankeado pronto, si no se encuentra ningún problema',
         'required_text' => 'Nominaciones: :current/:required',
+        'reset_at' => 'Nominations reset :time_ago by new problem :discussion.',
+        'reset_confirm' => 'Are you sure? Posting a new problem will reset nominations.',
```
Comment (ColdVolcano) [inline resources/lang/es/beatmaps.php]: ¿Pero que pasó aquí?

File: resources/lang/es/layout.php [hunk 5]
```diff
@@ -133,11 +141,12 @@
         ],
         'admin-store' => [
             '_' => 'admin::store',
-            'orders-index' => 'órdenes',
-            'orders-show' => 'órdenes',
+            'orders-index' => 'ordenes',
```
Comment (ColdVolcano) [inline resources/lang/es/layout.php]: No se bien el contexto, pero parece mas referirse a "órdenes" (de la tienda) que a alguien ordenando :p

File: resources/lang/es/model_validation/payments.php [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        'items' => [
+            'virtual_only' => 'El pago con `:provider` no es válido para productos físicos',
+        ],
+        'status' => [
+            'not_checkout' => 'Intentando aceptar el pago para una orden en mal estado `:state`.',
+            'not_paid' => 'Intentando reembolsar el pago para una orden en mal estado `:state`.',
+        ],
+    ],
+    'param' => [
+        'invalid' => 'El parámetro `:param` no coinciden',
```
Comment (ColdVolcano) [inline resources/lang/es/model_validation/payments.php]: "no coincide", o "Los parámetros"

File: resources/lang/es/users.php [hunk 7]
```diff
@@ -109,33 +135,31 @@
...
                     'accuracy' => 'precisión: :percentage',
-                    'title' => 'Jugadas Recientes',
+                    'title' => 'Jugadas Recientes (24h)',
                 ],
                 'title' => 'Historial',
             ],
             'kudosu' => [
                 'available' => 'Kudosu disponible',
-                'available_info' => 'Kudosu puede ser intercambiado por estrellas kudosu, que ayudarán a tu beatmap a obtener más atención. Este es el número de kudosu que no has intercambiado aún.',
+                'available_info' => 'Los kudosu pueden ser intercambiado por estrellas kudosu, que ayudarán a tu beatmap a obtener más atención. Este es el número de kudosu que no has intercambiado aún.',
```
Comment (ColdVolcano) [inline resources/lang/es/users.php]: intercambiado -> intercambiados

File: resources/lang/es/users.php [hunk 8]
```diff
@@ -109,33 +135,31 @@
...
                 'recent_entries' => 'Historial de Kudosu Reciente',
                 'title' => 'Kudosu!',
                 'total' => 'Kudosu Total Obtenido',
-                'total_info' => 'Basado en qué tanto ha colaborado el usuario a la moderación del beatmap. Ve <a href="'.osu_url('user.kudosu').'">esta página</a> para más información.',
-
+                'total_info' => 'Basado en qué tanto ha colaborado el usuario a la moderación del beatmap. Mira <a href="'.osu_url('user.kudosu').'">esta página</a> para más información.',
```
Comment (ColdVolcano) [inline resources/lang/es/users.php]: del beatmap -> de beatmaps

File: resources/lang/es/beatmaps.php [hunk 9]
```diff
@@ -38,17 +38,20 @@
         'deny_kudosu' => 'denegar kudosu',
         'edit' => 'editar',
         'edited' => 'Última edición por :editor :update_time',
+        'kudosu_denied' => 'Negado de obtener kudosu.',
         'message_placeholder' => 'Escribe aquí para publicar',
+        'message_placeholder_deleted_beatmap' => 'Esta dificultad ha sido eliminada así que ya no se puede discutirla más.',
         'message_type_select' => 'Seleccionar Tipo de Comentario',
```
Comment (ColdVolcano) [inline resources/lang/es/beatmaps.php]: 🐍

File: resources/lang/es/beatmaps.php (hunk 2 above)
Comment (Underforest) [reply resources/lang/es/beatmaps.php]: Fixed

File: resources/lang/es/beatmaps.php (hunk 1 above)
Comment (Underforest) [reply resources/lang/es/beatmaps.php]: Fixed

File: resources/lang/es/beatmaps.php (hunk 3 above)
Comment (Underforest) [reply resources/lang/es/beatmaps.php]: Whoops

File: resources/lang/es/layout.php (hunk 5 above)
Comment (Underforest) [reply resources/lang/es/layout.php]: Arreglado

File: resources/lang/es/beatmaps.php (hunk 4 above)
Comment (Underforest) [reply resources/lang/es/beatmaps.php]: Olvidé traducir esto lol

File: resources/lang/es/model_validation/payments.php (hunk 6 above)
Comment (Underforest) [reply resources/lang/es/model_validation/payments.php]: Arreglado como singular

File: resources/lang/es/users.php (hunk 7 above)
Comment (Underforest) [reply resources/lang/es/users.php]: Arreglado como plural

File: resources/lang/es/users.php (hunk 8 above)
Comment (Underforest) [reply resources/lang/es/users.php]: Arreglado

File: resources/lang/es/beatmaps.php (hunk 9 above)
Comment (Underforest) [reply resources/lang/es/beatmaps.php]: :rofl:

**Comment by nanaya** [comment]
Click on "Show all checks", click "Details" on travis ci build, find and expand `php artisan locale:check` line. Look for `Checking locale es`.

Extraneous (unused/renamed) keys:
- beatmapsets.show.scoreboard.list.accuracy
- beatmapsets.show.scoreboard.list.player-header
- beatmapsets.show.scoreboard.list.rank-header
- beatmapsets.show.scoreboard.list.score
- layout.footer.general.changelog
- layout.menu.admin.beatmapsets.covers

Missing keys:
- admin.beatmapsets.covers.regenerate
- admin.beatmapsets.covers.regenerating
- admin.beatmapsets.covers.remove
- admin.beatmapsets.covers.removing
- admin.beatmapsets.show.covers
- admin.users.restricted_banner.message
- admin.users.restricted_banner.title
- beatmapsets.show.details.favourited_count
- beatmapsets.show.scoreboard.headers.accuracy
- beatmapsets.show.scoreboard.headers.combo
- beatmapsets.show.scoreboard.headers.miss
- beatmapsets.show.scoreboard.headers.mods
- beatmapsets.show.scoreboard.headers.player
- beatmapsets.show.scoreboard.headers.pp
- beatmapsets.show.scoreboard.headers.rank
- beatmapsets.show.scoreboard.headers.score
- beatmapsets.show.scoreboard.headers.score_total
- beatmapsets.show.scoreboard.miss_count
- layout.footer.general.changelog-index
- model_validation.user.too_long

**Comment by Underforest** [comment]
Thanks for pointing it out! Gonna fix them later this day/week or so~

**Comment by Underforest** [comment]
According to Travis CI's check (lines 1291-1299), missing and weird lines are fixed and removed respectively. It should be OK right now.

**Comment by nekodex** [comment]
Thanks for the contribution!

I'm gonna merge this for now - there's 3 missing translations (that were added since this PR was made):
```
Missing keys:
- beatmapsets.show.info.genre
- beatmapsets.show.info.language
- common.buttons.see_more
```
...and a few new ones coming soon in some PRs, so let's merge this before it gets even more out of date.

Feel free to open a new PR with the missing translations if you want. 😃

**Pull request merged**